tensorflow/python/kernel_tests/rnn_test.py

```diff
@@ -574,59 +574,16 @@ def _testDynamicEquivalentToStaticRNN(self, use_gpu, use_sequence_length):
       sequence_length = None
 
     ########### Step 1: Run static graph and generate readouts
-    with self.test_session(use_gpu=use_gpu, graph=tf.Graph()) as sess:
-      concat_inputs = tf.placeholder(tf.float32,
-                                     shape=(time_steps, batch_size, input_size))
-      inputs = tf.unpack(concat_inputs)
-      initializer = tf.random_uniform_initializer(-0.01, 0.01, seed=self._seed)
-
-      cell = tf.nn.rnn_cell.LSTMCell(
-          num_units, input_size, use_peepholes=True,
-          initializer=initializer, num_proj=num_proj)
-
-      with tf.variable_scope("dynamic_scope"):
-        outputs_static, state_static = tf.nn.rnn(
-            cell, inputs, sequence_length=sequence_length, dtype=tf.float32)
-
-      feeds = {concat_inputs: input_values}
-
-      # Initialize
-      tf.initialize_all_variables().run(feed_dict=feeds)
-
-      # Generate gradients of sum of outputs w.r.t. inputs
-      static_gradients = tf.gradients(
-          outputs_static + [state_static], [concat_inputs])
-
-      # Generate gradients of individual outputs w.r.t. inputs
-      static_individual_gradients = _flatten([
-          tf.gradients(y, [concat_inputs])
-          for y in [outputs_static[0],
-                    outputs_static[-1],
-                    state_static]])
-
-      # Generate gradients of individual variables w.r.t. inputs
-      trainable_variables = tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES)
-      assert len(trainable_variables) > 1, (
-          "Count of trainable variables: %d" % len(trainable_variables))
-      # pylint: disable=bad-builtin
-      static_individual_variable_gradients = _flatten([
-          tf.gradients(y, trainable_variables)
-          for y in [outputs_static[0],
-                    outputs_static[-1],
-                    state_static]])
-
-      # Test forward pass
-      values_static = sess.run(outputs_static, feed_dict=feeds)
-      (state_value_static,) = sess.run((state_static,), feed_dict=feeds)
-
-      # Test gradients to inputs and variables w.r.t. outputs & final state
-      static_grad_values = sess.run(static_gradients, feed_dict=feeds)
-
-      static_individual_grad_values = sess.run(
-          static_individual_gradients, feed_dict=feeds)
+    initializer = tf.random_uniform_initializer(-0.01, 0.01, seed=self._seed)
+    cell = tf.nn.rnn_cell.LSTMCell(
+        num_units, input_size, use_peepholes=True,
```

Comment:
thats existing code. for tensorflow/python/kernel_tests/rnn_test.py
<https://github.com/tensorflow/tensorflow/pull/2002#discussion_r62930648> I
only refactored some of the benchmarking code so I can leverage it in my
contrib/../rnn_test.py

--
William Chan
Carnegie Mellon University
(650) 450-9455
williamchan.ca

On Wed, May 11, 2016 at 2:32 PM, ebrevdo <notifications@github.com> wrote:

> In tensorflow/python/kernel_tests/rnn_test.py
> <https://github.com/tensorflow/tensorflow/pull/2002#discussion_r62930648>:
>
> > -          for y in [outputs_static[0],
> > -                    outputs_static[-1],
> > -                    state_static]])
> > -
> > -      # Test forward pass
> > -      values_static = sess.run(outputs_static, feed_dict=feeds)
> > -      (state_value_static,) = sess.run((state_static,), feed_dict=feeds)
> > -
> > -      # Test gradients to inputs and variables w.r.t. outputs & final state
> > -      static_grad_values = sess.run(static_gradients, feed_dict=feeds)
> > -
> > -      static_individual_grad_values = sess.run(
> > -          static_individual_gradients, feed_dict=feeds)
> > +    initializer = tf.random_uniform_initializer(-0.01, 0.01, seed=self._seed)
> > +    cell = tf.nn.rnn_cell.LSTMCell(
> > +        num_units, input_size, use_peepholes=True,
>
> here you have use_peepholes = true. are you sure you want that?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/2002/files/fc40971bba82762fe413b5e7d6fe12a09722876b#r62930648>
>